editor/editor_settings.cpp

```diff
@@ -849,10 +848,16 @@ void EditorSettings::create() {
 			goto fail;
 		}
 
-		singleton = ResourceLoader::load(config_file_path, "EditorSettings");
+		if (!p_use_default_settings) {
+			singleton = ResourceLoader::load(config_file_path, "EditorSettings");
+		}
 
 		if (singleton.is_null()) {
-			ERR_PRINT("Could not load editor settings from path: " + config_file_path);
+			if (!p_use_default_settings) {
+				// This is expected behavior when forcing the use of default settings (used by `--doctool`),
+				// so don't print an error message about it.
+				ERR_PRINT("Could not load editor settings from path: " + config_file_path);
+			}
```

Comment:
That doesn't really make sense to me, just above you're explicitly avoiding the creation of `singleton`, so obviously it's going to be `null`. You should just skip both:
```cpp
if (!p_use_default_settings) {
	singleton = ResourceLoader::load(config_file_path, "EditorSettings");

	if (singleton.is_null()) {
		ERR_PRINT("Could not load editor settings from path: " + config_file_path);
		goto fail;
	}
}
```

But then I don't get how `EditorSettings::get_singleton()` works in DocTools...

Reply:
Oh I see, that's super hacky... you're relying on `goto fail;` from an unrelated error to get a default initialized `singleton`...